src/audio/module_adapter/module_adapter.c

```diff
@@ -240,9 +241,17 @@ static int module_adapter_dp_queue_prepare(struct comp_dev *dev)
 			 &sink_buffer->stream.runtime_stream_params,
 			 sizeof(sink_buffer->stream.runtime_stream_params));
 		/* calculate time required the module to provide OBS data portion - a period */
-		unsigned int sink_period = 1000000 * sink_get_min_free_space(mod->sinks[i]) /
+		uint32_t min_free_size =
+			sink_get_frame_bytes(mod->sinks[i]) *
+			(sink_get_rate(mod->sinks[i]) / (USEC_PER_SEC / LL_TIMER_PERIOD_US));
+		min_free_size = MIN(min_free_size, sink_get_min_free_space(mod->sinks[i]));
+
+		unsigned int sink_period = 1000000 * min_free_size /
```

Comment:
can this overflow? If `min_free_size` is a bit more than 4K, it will overflow already?

Reply:
I think there is no case where the minimal size will be close to 4K value.

Reply:
take 192khz, 32 bit 8 ch with 10ms period data as example:
192 * 4 * 8 * 10 = 61440, even 1ms data size is 6144.

Reply:
this is in prepare, so it is one time, no worry on performance.